pkg/sql/randgen/mutator.go

```diff
@@ -847,6 +822,42 @@ func postgresCreateTableMutator(
 	return mutated, changed
 }
 
+// replaceNonImmutableExpr checks if expr has a non-immutable operation in it,
+// and returns an immutable expr if it does. This is needed because Postgres has
+// different cast volatility for timestamps and/OID types. The substitution here
+// is specific to the output of randgen.randExpr, which uses
+// `lower(cast(col as string))`.
+func replaceNonImmutableExpr(
+	rng *rand.Rand, expr tree.Expr, colTypes map[string]*types.T,
+) (newExpr tree.Expr, changed bool) {
+	if funcExpr, ok := expr.(*tree.FuncExpr); ok {
+		if len(funcExpr.Exprs) == 1 {
+			if castExpr, ok := funcExpr.Exprs[0].(*tree.CastExpr); ok {
+				referencedType := colTypes[castExpr.Expr.(*tree.UnresolvedName).String()]
+				isContextDependentType := referencedType.Family() == types.TimestampFamily ||
```

Comment:
can we make `lookupCast` in the sql package public to avoid `contextdDpendenType` by instead checking for volatility?

Reply:
well we could - but the problem is that our `lookupCast` will return a different volatility than postgres. actually, randgen already uses `lookupCast`. https://github.com/cockroachdb/cockroach/blob/e89328d92398a3e2d6487179845a51e7f1caa435/pkg/sql/randgen/expr.go#L215

we could make `lookupCast` have a `matchPostgresVolatility` flag, but that's sketchy to add in production code.